plugin/mutate/source/dextool/plugin/mutate/backend/report/html/page_tree_map.d

```diff
@@ -0,0 +1,100 @@
+module dextool.plugin.mutate.backend.report.html.page_tree_map;
+
+import std.json : JSONValue;
+import arsd.dom : Document, Element, require, Table, RawSource;
+
+import dextool.plugin.mutate.backend.report.html : FileIndex;
+import dextool.plugin.mutate.backend.report.html.tmpl : tmplBasicPage;
+
+/* 
+* A JSON-like object that is more easily manipulated than JSONValue.
+* Has operator overloads for "index", "index assign" and "in" allowing the children list to be private
+* and cleaner construction of the object. 
+*/
+class JSONLike {
+    string name;
+    private JSONLike[string] children;
+    long locs = -1;
+    double score = -1;
+
+    JSONValue toJSONValue() {
+        JSONValue root = ["name" : name];
+
+        if (children.length) {
+            JSONValue[] descendants = [];
+            foreach (c; children) {
+                descendants ~= c.toJSONValue();
+            }
+            root.object["children"] = JSONValue(descendants);
+        }
+        if (locs != -1)
+            root.object["locs"] = JSONValue(locs);
+
+        if (score != -1)
+            root.object["score"] = JSONValue(score);
+
+        return root;
+    }
+
+    JSONLike opIndex(string i) pure nothrow {
+        return children[i];
+    }
+
+    JSONLike opIndexAssign(JSONLike value, string i) pure nothrow {
+        children[i] = value;
+        return value;
+    }
+
+    bool opBinaryRight(string op)(string a) {
+        if (op == "in") {
+            if (a in children)
+                return true;
+            return false;
+        }
+        return false;
+    }
+
+}
+
+@trusted auto makeTreeMapJSON(FileIndex[] files) {
+    import std.array : array;
+    import std.path : pathSplitter;
+
+    auto iRoot = new JSONLike();
+    iRoot.name = "root";
+
+    foreach (f; files) {
+        auto path = pathSplitter(f.display);
+        auto parent = iRoot;
+        foreach (seg; path) {
+            if (seg !in parent) {
+                auto child = new JSONLike();
+                child.name = seg;
+                parent[seg] = child;
+            }
+            parent = parent[seg];
+        }
+        parent.locs = f.totalMutants;
+        if (f.totalMutants == 0)
+            parent.score = 1.0;
+        else
+            parent.score = cast(double) f.killedMutants / cast(double) f.totalMutants;
+    }
+    return iRoot.toJSONValue().toPrettyString();
+}
+
+auto makeTreeMapPage(FileIndex[] files) {
```

Comment:
I would recommend re-arranging the source code so the public symbols are exposed at the top of the file. Then it is easy to add a blanket `private:` where the internal symbols start.
I think, in this case, it is only this function that is meant to be public so a "private:" after this function should do it?